File: 백준/Silver/6550. 부분 문자열/부분 문자열.py
import sys
input = sys.stdin.readline

while True:
    try:
        s,t = input().split()
        
        s_idx = 0
        cnt = 0 
        flag = False
        for i in range(len(t)):
            if t[i] == s[s_idx]:
                s_idx += 1
                cnt += 1
            if cnt == len(s):
                flag = True
                break
        
        if flag:
            print("Yes")
        else:
            print("No")
    
    except:
        break

File: 백준/Silver/6550. 부분 문자열/부분 문자열.js
const fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

for (let i=0; i<input.length; i++) {
  const [s,t] = input[i].trim().split(' ');
  
  const sQueue = s.split("");
  
  for (target of t) {
    if (target === sQueue[0]) {
      sQueue.shift();
    }
  }

  console.log(sQueue.length === 0 ? 'Yes' : 'No')
}